tests/conftest.py

```diff
@@ -293,6 +293,7 @@ def __init__(self, *args, **kwargs):
             self.plugin_manager = mocker.MagicMock()
             self.repodata_fns = ["repodata.json", "current_repodata.json"]
             self.subdir = mocker.MagicMock()
+            self.channel_settings = []
```

Comment:
It would maybe be better if the `MockContext` more closely resembled the context object. That way changes in downstream dependencies would not cause these types of errors. Right now, this bug is causing test failures on the main branch, so I think this quick fix is appropriate. If there is interest, I can spin out an issue to dig into a better solution.